src/sentry/eventstream/kafka/backend.py

```diff
@@ -252,14 +254,62 @@ def relay(self, consumer_group, commit_log_topic,
             initial_offset_reset=initial_offset_reset,
         )
 
-        consumer.subscribe([self.publish_topic])
+        owned_partition_offsets = {}
+
+        def on_assign(consumer, partitions):
+            updated_assignment = {(i.topic, i.partition): i.offset for i in partitions}
+
+            # Handle newly assigned partitions by saving them in the owned
+            # partition offsets map.
+            for topic, partition in set(updated_assignment) - set(owned_partition_offsets):
+                logger.debug('Received new partition assignment: %r', (topic, partition))
+
+                offset = updated_assignment[(topic, partition)]
+                if offset is OFFSET_INVALID:
+                    offset = None
+                else:
+                    assert offset > 0, 'unexpected negative offset'
```

Comment:
Can't think of any situation where this would be negative and anything other than `OFFSET_INVALID` so just trying to be extra careful.